includes/payments/class-edd-payment.php

```diff
@@ -2212,18 +2232,21 @@ public function update_meta( $meta_key = '', $meta_value = '', $prev_value = ''
 									'amount' => $this->subtotal - $discount->get_discounted_amount( $this->subtotal ),
 								) );
 
+								continue;
+							}
+
 							// Add the discount as an adjustment.
-							} else {
-								edd_add_order_adjustment( array(
+							edd_add_order_adjustment(
```

Comment:
I prefer the if/else approach. I find it easier to read. This whole method is so massive that it's easy to lose track of what the `continue` is actually for.